src/ApiService/ApiService/Functions/AgentEvents.cs

```diff
@@ -110,18 +110,23 @@ public class AgentEvents {
             _log.LogInformation("node now available for work: {MachineId}", machineId);
         } else if (ev.State == NodeState.SettingUp) {
             if (ev.Data is NodeSettingUpEventData settingUpData) {
-                if (!settingUpData.Tasks.Any()) {
+                if (settingUpData.Tasks?.Any() == false || settingUpData.TaskData?.Any() == false) {
                     return Error.Create(ErrorCode.INVALID_REQUEST,
                         $"setup without tasks.  machine_id: {machineId}"
                     );
                 }
 
-                foreach (var taskId in settingUpData.Tasks) {
-                    var task = await _context.TaskOperations.GetByTaskId(taskId);
+                var tasks =
+                    settingUpData.Tasks is not null
+                    ? settingUpData.Tasks.Select(t => (t, _context.TaskOperations.GetByTaskIdSlow(t)))
+                    : settingUpData.TaskData!.Select(t => (t.TaskId, _context.TaskOperations.GetByJobIdAndTaskId(t.JobId, t.TaskId)));
```

Comment:
TaskData is not null <- is that 100% ?

Reply:
We require either Tasks (old) or TaskData (new) to be set. It's not guaranteed by the schema but the agent code will be sending one or the other, we don't have external clients submitting this.